client/src/components/category/CategoryForm.js

```diff
@@ -0,0 +1,87 @@
+import React, { useState, useEffect } from "react";
+import Api from "../../Api";
+import { useHistory, useParams } from "react-router-dom";
+
+function CategoryForm() {
+  const history = useHistory();
+  const { id } = useParams();
+  const [category, setCategory] = useState({
+    categoryName: "",
+    limit: 0.0,
+    spend: 0.0,
+  });
+
+  useEffect(() => {
+    if (id) {
+      Api.categories.get(id).then((response) => setCategory(response.data));
+    }
+  }, [id]);
+
+  const onChange = (e) => {
+    const { name, value } = e.target;
+    const newCategory = { ...category };
+    if (value === "") {
+      newCategory[name] = value;
+    } else {
+      newCategory[name] = parseFloat(value);
```

Comment:
This code is causing editing the name to fail, because once you start entering text, it trys to parse it into a float.

I'm not even sure this is entirely necessary- the DB should be able to convert from string to decimal as needed...